### PR TITLE
introduce nit_env.sh to setup the shell environement

### PR DESCRIPTION
**Pull request by privat**
The script `nit_env.sh` tries to auto-magically configure PATH, MANPATH and bash completion for users. The point is to be as portable and simple a possible for the final used has he just has to write

~~~
$ source misc/nit_env.sh
~~~

and get a working setup.

Moreover, if `install` in given as argument, then the script register itself to the user `$HOME/.profile`.

~~~
$ source misc/nit_env.sh install
~~~

One advantage is that the script invocation is registered in `.profile`, thus future evolutions of the script will be automatically used in future sessions of the users.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,11 @@ NITCOPT=
 # Additional program directories (contrib and examples) that are buildable
 PROGS=$(dir $(wildcard examples/*/Makefile contrib/*/Makefile))
 
-all: tools
+all: tools man
+	@echo ""
+	@echo "Congratulatin, Nit was succesfully compiled."
```
Comment (xymus) [inline Makefile]: *Congratulations

**Comment by xymus** [comment]
+1

File: misc/nit_env.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Source this script inside a bash or sh session to setup:
+#
+# * PATH
+# * MANPATH
+# * bash_completion
+#
+# If `install` is given as an argument, then the script sourcing is automatically inserted in the `.profile`
```
Comment (Morriar) [inline misc/nit_env.sh]: Missing the sentence final dot.

File: misc/nit_env.sh [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+# Final function that setup the shell environment.
```
Comment (Morriar) [inline misc/nit_env.sh]: setups*

**Comment by Morriar** [comment]
+1

File: misc/nit_env.sh (hunk 3 above)
Comment (privat) [inline misc/nit_env.sh]: For what is worth: http://notaverb.com/setup

**Comment by privat** [comment]
reroll: more doc and robustness

File: misc/nit_env.sh (hunk 3 above)
Comment (Morriar) [inline misc/nit_env.sh]: Hum! Good to know!

**Pull request merged**